From 4eef3346fdc6b8fbbc205a3d8949101517892593 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Fri, 18 May 2018 13:18:50 +0700
Subject: [PATCH] mmu_common: fix fallthrough compile issue

Compile issue "error: this statement may fall through [-Werror=implicit-fallthrough=]"
Fixed by adding fallthrough attribute.

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 services/server/common/mmu_common.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/services/server/common/mmu_common.c b/services/server/common/mmu_common.c
index 1627975..ce10306 100755
--- a/services/server/common/mmu_common.c
+++ b/services/server/common/mmu_common.c
@@ -1498,6 +1498,7 @@ e0:
 						/* Check we haven't wrapped around */
 						PVR_ASSERT(psLevel->ui32RefCount <= psLevel->ui32NumOfEntries);
 					}
+					__attribute__ ((fallthrough));
 			case 2:
 					if (psLevel->apsNextLevel[i] != IMG_NULL  &&
 					    psLevel->apsNextLevel[i]->ui32RefCount == 0)
@@ -1505,6 +1506,7 @@ e0:
 						_PxMemFree(psMMUContext, &psLevel->sMemDesc,
 									aeMMULevel[uiThisLevel]);
 					}
+					__attribute__ ((fallthrough));
 			case 1:
 					if (psLevel->apsNextLevel[i] != IMG_NULL  &&
 					    psLevel->apsNextLevel[i]->ui32RefCount == 0)
@@ -1512,6 +1514,7 @@ e0:
 						OSFreeMem(psLevel->apsNextLevel[i]);
 						psLevel->apsNextLevel[i] = IMG_NULL;
 					}
+					__attribute__ ((fallthrough));
 			case 0:
 					uiAllocState = 3;
 					break;
@@ -1828,7 +1831,7 @@ static IMG_VOID _MMU_GetPTEInfo(MMU_CONTEXT *psMMUContext,
 			/* find the page directory containing the PCE */
 			uiPCEIndex = _CalcPCEIdx(sDevVAddr, psDevVAddrConfig, IMG_FALSE);
 			psLocalLevel = psMMUContext->sBaseLevelInfo.apsNextLevel[uiPCEIndex];
-
+			__attribute__ ((fallthrough));
 		case MMU_LEVEL_2:
 			/* find the page table containing the PDE */
 			uiPDEIndex = _CalcPDEIdx(sDevVAddr, psDevVAddrConfig, IMG_FALSE);
@@ -1840,7 +1843,7 @@ static IMG_VOID _MMU_GetPTEInfo(MMU_CONTEXT *psMMUContext,
 			{
 				psLocalLevel = psMMUContext->sBaseLevelInfo.apsNextLevel[uiPDEIndex];
 			}
-
+			__attribute__ ((fallthrough));
 		case MMU_LEVEL_1:
 			/* find PTE index into page table */
 			*pui32PTEIndex = _CalcPTEIdx(sDevVAddr, psDevVAddrConfig, IMG_FALSE);
-- 
2.7.4

